confluent_kafka/avro/__init__.py

```diff
@@ -29,23 +29,17 @@ class AvroProducer(Producer):
     def __init__(self, config, default_key_schema=None,
                  default_value_schema=None, schema_registry=None):
 
-        schema_registry_url = config.pop("schema.registry.url", None)
-        schema_registry_ca_location = config.pop("schema.registry.ssl.ca.location", None)
-        schema_registry_certificate_location = config.pop("schema.registry.ssl.certificate.location", None)
-        schema_registry_key_location = config.pop("schema.registry.ssl.key.location", None)
+        sr_conf = {key.replace("schema.registry.", ""): value
+                   for key, value in config.items() if key.startswith("schema.registry.") or key.startswith("sasl")}
```

Comment:
This will just result in a dict with a few unused keys, once it falls out of the constructor's scope it will be eligible for garbage collection and no longer a problem no?

Realistically I could just pass the entire dict to the SR client if you'd prefer.

Reply:
But the SR client shouldn't silently ignore configs it doesn't understand though, but error out on them.

Reply:
Fixed, including a check for unrecognized properties